dev/tools/localization/bin/gen_date_localizations.dart

```diff
@@ -180,14 +178,14 @@ Set<String> _supportedLocales() {
   for (final FileSystemEntity entity in supportedLocalesDirectory.listSync()) {
     final String filePath = entity.path;
     if (FileSystemEntity.isFileSync(filePath) && filenameRE.hasMatch(filePath)) {
-      supportedLocales.add(filenameRE.firstMatch(filePath)[1]);
+      supportedLocales.add(filenameRE.firstMatch(filePath)![1]!);
     }
   }
 
   return supportedLocales;
 }
 
-Map<String, File> _listIntlData(Directory directory) {
+Map<String, File?> _listIntlData(Directory directory) {
```

Comment:
Will this actually ever return `null` for a File? My guess would be "no". If that's correct, we should refactor this method to make `File` non-nullable.